extensions/ql-vscode/src/authentication.ts

```diff
@@ -13,33 +13,38 @@ const SCOPES = ['repo'];
 export class Credentials {
   private octokit: Octokit.Octokit | undefined;
 
-  async initialize(context: vscode.ExtensionContext): Promise<void> {
-    this.registerListeners(context);
-    this.setOctokit();
+  // Explicitly make the constructor private, so that we can't accidentally call the constructor from outside the class
+  // without also initializing the class.
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() { }
+
+  static async initialize(context: vscode.ExtensionContext): Promise<Credentials> {
+    const c = new Credentials();
+    c.registerListeners(context);
+    await c.initializeOctokit(false);
+    return c;
   }
 
-  private async setOctokit() {
-    // If `createIfNone` were true, a dialog would pop up asking the user to authenticate as soon as the extension starts.
-    // Setting `createIfNone` to false for now, so we can have a more "quiet" prompt, i.e. a numbered label on the "accounts"
-    // icon in the activity bar.
-    const session = await vscode.authentication.getSession(GITHUB_AUTH_PROVIDER_ID, SCOPES, { createIfNone: false });
+  private async initializeOctokit(createIfNone: boolean) {
+    // If `createIfNone` is true, a dialog pops up asking the user to authenticate as soon as the extension starts.
+    // Initializing with `createIfNone: false` for now, so we can have a more quiet prompt, i.e. a numbered label on
+    // the "accounts" icon in the activity bar.
+    const session = await vscode.authentication.getSession(GITHUB_AUTH_PROVIDER_ID, SCOPES, { createIfNone: createIfNone });
 
     if (session) {
       this.octokit = new Octokit.Octokit({
         auth: session.accessToken
       });
-
-      return;
+    } else {
+      this.octokit = undefined;
     }
```

Comment:
Here, instead of setting the variable, you return:

```suggestion
    return session
      ? new Octokit.Octokit({
        auth: session.accessToken
      }) : undefined;
```